src/Components/Scenes/Hydration/App/Transform/TransformEditor.tsx

```diff
@@ -105,12 +113,18 @@ const TransformEditor = (props: TransformProps) => {
       <div className="Editor__heading" />
       <div className={classes.name}>
         {isScriptNameActive ? (
-          <Field
+          <TextField
             autoFocus
-            onBlur={() => setIsScriptNameActive(!isScriptNameActive)}
+            onBlur={() => {
+              setChange(scriptTitle, 'name');
+              return setIsScriptNameActive(!isScriptNameActive);
+            }}
+            onChange={e => {
+              dispatch(handleTitleName(e.target.value));
+            }}
             name={`transforms.${id}.name`}
             className={classes.scriptName}
-            as={TextField}
+            value={scriptTitle}
```

Comment:
This should actually be setting the `placeholder` property instead of the `value` property.

Reply:
Updated to `placeholder` property